tests/pytests/unit/test_master.py

```diff
@@ -4,7 +4,49 @@
 
 import salt.master
 import salt.utils.platform
-from tests.support.mock import patch
+from tests.support.mock import MagicMock, patch
+
+
+@pytest.fixture
+def master_opts():
+    """
+    Master options
+    """
+    yield salt.config.master_config(None)
+
+
+@pytest.fixture
+def maintenence_opts(master_opts):
+    """
+    Options needed for master's Maintenence class
+    """
+    opts = master_opts.copy()
+    opts.update(git_pillar_update_interval=180, maintenance_interval=181)
+    yield opts
+
+
+@pytest.fixture
+def maintenence(maintenence_opts):
+    """
+    The master's Maintenence class
+    """
+    maintenence = salt.master.Maintenance(maintenence_opts)
+    try:
+        yield maintenence
+    finally:
+        pass
```

Comment:
```suggestion
@pytest.fixture
def maintenence(maintenence_opts):
    """
    The master's Maintenence class
    """
    return salt.master.Maintenance(maintenence_opts)
```
Since you're not doing anything once the test finishes.